gtsam/sfm/sfm.i

```diff
@@ -252,14 +262,21 @@ class MFAS {
 };
 
 #include <gtsam/sfm/TranslationRecovery.h>
+class TranslationRecoveryParams {
+  gtsam::LevenbergMarquardtParams lmParams;
+  gtsam::Values initial;
+};
+
 class TranslationRecovery {
   TranslationRecovery(
       const gtsam::BinaryMeasurementsUnit3& relativeTranslations,
-      const gtsam::LevenbergMarquardtParams& lmParams);
-  TranslationRecovery(
-      const gtsam::BinaryMeasurementsUnit3&
-          relativeTranslations);  // default LevenbergMarquardtParams
-  gtsam::Values run(const double scale) const;
+      const gtsam::TranslationRecoveryParams& lmParams);
+  TranslationRecovery(const gtsam::BinaryMeasurementsUnit3&
+                          relativeTranslations);  // default params
+  gtsam::Values run(const gtsam::BinaryMeasurementsPoint3& betweenTranslations,
+                    const double scale) const;
+  // default empty betweenTranslations
+  // gtsam::Values run(const double scale) const;
```

Comment:
@dellaert is there a way to wrap the method this way? using the default value of the first betweenTranslations argument?

Reply:
Actually, I think run should have *both* `relativeTranslations` and `betweenTranslations`. The constructor should just take parameters, not data, so you can "run" multiple datasets.
If you agree, you could make scale the first argument and provide two declarations in the .i file:

```c++
gtsam::Values run(const double scale, const gtsam::BinaryMeasurementsUnit3& relativeTranslations) const;
gtsam::Values run(const double scale, const gtsam::BinaryMeasurementsUnit3& relativeTranslations, const gtsam::BinaryMeasurementsPoint3& betweenTranslations) const;
```